File: android/takanawa-android/src/main/kotlin/ai/yetanother/takanawa/DownloadPhase.kt
package ai.yetanother.takanawa

enum class DownloadPhase(val code: Int) {
    CREATED(0),
    RUNNING(1),
    PAUSED(2),
    CANCELLED(3),
    COMPLETED(4),
    FAILED(5),
    PAUSING(6),
    CANCELLING(7),
    ;

    companion object {
        @JvmStatic
        fun fromCode(code: Int): DownloadPhase =
            values().firstOrNull { it.code == code } ?: FAILED
    }
}

File: packages/takanawa-csharp/src/YetAnotherAI.Takanawa/DownloadPhase.cs
namespace YetAnotherAI.Takanawa
{
    public enum DownloadPhase
    {
        Created = 0,
        Running = 1,
        Paused = 2,
        Cancelled = 3,
        Completed = 4,
        Failed = 5,
        Pausing = 6,
        Cancelling = 7,
        Starting = 8,
        Allocating = 9,
        Verifying = 10,
    }
}
